### ISSUE TITLE
[V2][iOS] topBar.title.fontFamily reset on mergeOptions

### ISSUE THREAD

**Issue by Jazqa** [wix/react-native-navigation#4249]
### Issue Description

topBar.title.fontFamily is being reset with mergeOptions

### Steps to Reproduce / Code Snippets / Screenshots

Have a topBar.title.fontFamily set in defaultOptions and try to dynamically adjust the topBar by using mergeOptions. The fontFamily set in the defaultOptions will not be used.

---
### Environment
* React Native Navigation version: 2.0.2628
* React Native version: 0.56
* Platform(s) (iOS, Android, or both?): iOS

**Comment by Jazqa** [comment]
That's unrelated in my case, since I am not using a subtitle

**Comment by pkyeck** [comment]
yep, but as I said in the comment in 4188, for me it was not related to using a subtitle - happened also when just trying to overwrite the fontFamily for title ...

**Comment by pkyeck** [comment]
this is still a problem afaik

**Comment by zabojad** [comment]
I'm having the same issue with `react-native-navigation 2.2.5`.

It seems that `Navigation.mergeOptions()` does not perform a deep merge, as simple as that.

**Comment by tyrong** [comment]
Looks to be true, seems to be doing a lodash deepClone, but anything deeper than one level like `topbar: {
    background: {
          color: '#fff'
     }
}`
doesnt work, any solution to this ?

**Comment by zabojad** [comment]
@tyrong  I don't have any solution yet.

In my case, I was trying to change the topbar title text after the screen shows up with a hidden topbar (`visible: false` option).

So I was doing:
``` 
Navigation.mergeOptions(props.componentId,{ topBar: { visible: true, title: { text: "blablabla..." }}});
```
and was loosing all my topBar default options.

I tried to workaround this problem by reseting at the same time all default options. It almost worked except for the `alignment: 'center'` option (Android) that I'm still loosing in that case...

The deep merge is perhaps one thing but it might not be the only issue with mergeOptions for topBar...

**Comment by zabojad** [comment]
@guyca I'm having that on Android (I see you've tagged this issue as an iOS one).

**Comment by dsaltares** [comment]
This is not stale. I can still reproduce.

**Comment by Kouznetsov** [comment]
[Hateful anti stale bot comment stating that I also have this issue]
